tooling/autodispose-error-prone-checker/src/main/java/com/uber/autodispose/error/prone/checker/MissingAutoDisposeErrorChecker.java

```diff
@@ -0,0 +1,175 @@
+        .add(instanceMethod().onDescendantOf("io.reactivex.Maybe").named(SUBSCRIBE))
+        .build();
+  }
+
+  /**
+   * Matcher to find the as operator in the observable chain.
+   */
+  private static final Matcher<ExpressionTree> METHOD_NAME_MATCHERS =
+      new Matcher<ExpressionTree>() {
+        @Override
+        public boolean matches(ExpressionTree tree, VisitorState state) {
+          if (!(tree instanceof MethodInvocationTree)) {
+            return false;
+          }
+          MethodInvocationTree invTree = (MethodInvocationTree) tree;
+
+          final MemberSelectTree memberTree = (MemberSelectTree) invTree.getMethodSelect();
+          if (!memberTree.getIdentifier().contentEquals(AS)) {
+            return false;
+          }
+
+          for (MethodMatchers.MethodNameMatcher nameMatcher : AS_CALL_MATCHERS) {
+            if (nameMatcher.matches(invTree, state)) {
+              ExpressionTree arg = invTree.getArguments().get(0);
+              final Type scoper = state
+                  .getTypeFromString("com.uber.autodispose.AutoDisposeConverter");
+              return ASTHelpers.isSubtype(ASTHelpers.getType(arg), scoper, state);
+            }
+          }
+          return false;
+        }
+      };
+
+  private static Matcher<MethodInvocationTree> matcher(List<String> classesWithLifecycle) {
+    return new Matcher<MethodInvocationTree>() {
+      @Override
+      public boolean matches(MethodInvocationTree tree, VisitorState state) {
+
+        boolean matchFound = false;
+        final MemberSelectTree memberTree = (MemberSelectTree) tree.getMethodSelect();
+        if (!memberTree.getIdentifier().contentEquals(SUBSCRIBE)) {
+          return false;
+        }
+
+        for (MethodMatchers.MethodNameMatcher nameMatcher : SUBSCRIBE_MATCHERS) {
+          if (!nameMatcher.matches(tree, state)) {
+            continue;
+          } else {
+            matchFound = true;
+            break;
+          }
+        }
+        if (!matchFound) {
+          return false;
+        }
+
+        ClassTree enclosingClass =
+            ASTHelpers.findEnclosingNode(state.getPath(), ClassTree.class);
+        Type.ClassType enclosingClassType = ASTHelpers.getType(enclosingClass);
+
+        for (String s : classesWithLifecycle) {
+          Type lifecycleType = state.getTypeFromString(s);
+          if (ASTHelpers.isSubtype(enclosingClassType, lifecycleType, state)) {
+            return !METHOD_NAME_MATCHERS.matches(memberTree.getExpression(), state);
+          }
+        }
+        return false;
+      }
+    };
+  }
+
+  @Override
+  public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState state) {
+    if (matcher.matches(tree, state)) {
+      return buildDescription(tree).build();
+    } else {
+      return Description.NO_MATCH;
+    }
+  }
+
+  @Override
+  public String linkUrl() {
+    return "http://t.uber.com/autodispose-leak-checker";
```

Comment:
Let's link something public :). `https://github.com/uber/AutoDispose/wiki/Error-Prone-Checker`. I'll flesh that wiki page out separately after this lands.

Reply:
https://github.com/uber/AutoDispose/pull/156/commits/2547c2fbb20ad46adb32b4cb7d2da8b605dcf89e